### PR TITLE
Better Meta/OG descriptions

### PR DESCRIPTION
**Pull request by jpaddison3**
On slack in particular, our  descriptions being way too long makes our previews look pretty bad.

I heard from user interviews that posts being shared in slack was an important way that people were visiting the Forum. So now we have shorter previews.

A particular problem with our previews is that we don't do anything for paragraph breaks except for a space. So we cut posts off based on paragraphs.

Sam and I went a little "extra" and cut out the first paragraph from the description if it was just an epistemic status or crossposting notice.

### CODE REVIEW

File: packages/lesswrong/server/scripts/testPostDescription.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/* eslint-disable no-console */
+import { getPostDescription } from "../../components/posts/PostsPage/PostsPage";
+import { Posts } from "../../lib/collections/posts";
+import Revisions from "../../lib/collections/revisions/collection";
+import { Vulcan } from "../vulcan-lib";
+
+
+/** For visually inspecting that our descriptions match the post content well */
+const run = async () => {
```
Comment (jpaddison3) [inline packages/lesswrong/server/scripts/testPostDescription.ts:9]: You're welcome to have a go if you want:

```
❯ ./scripts/serverShellCommand.sh 'Globals.testPostDescription()'
```

Comment (s-cheng) [reply packages/lesswrong/server/scripts/testPostDescription.ts:9]: Nice!

File: packages/lesswrong/components/posts/PostsPage/PostsPage.tsx [hunk 2]
```diff
@@ -13,6 +13,36 @@ import { viewNames } from '../../comments/CommentsViews';
 
 export const MAX_COLUMN_WIDTH = 720
 
+const POST_DESCRIPTION_EXCLUSIONS: RegExp[] = [/crossposted/i, /epistemic status/i];
```
Comment (s-cheng) [inline packages/lesswrong/components/posts/PostsPage/PostsPage.tsx:16]: When I ran the test script, I did see one with "(Cross-posted on The Life You Can Save, Intentional Insights, and LessWrong)." go through, so maybe also include `cross-posted`

File: packages/lesswrong/components/posts/PostsPage/PostsPage.tsx [hunk 3]
```diff
@@ -13,6 +13,36 @@ import { viewNames } from '../../comments/CommentsViews';
...
+    // concatenate the first few paragraphs together up to some reasonable length
+    const firstFewPars = post.contents.plaintextDescription
+      // paragraphs in the plaintext description are separated by double-newlines
+      .split(/\n\n/)
+      // get rid of bullshit opening text ('epistemic status' or 'crossposted from' etc)
+      .filter((par) => !POST_DESCRIPTION_EXCLUSIONS.some((re) => re.test(par)))
+      // concatenate paragraphs together with a delimiter, until they reach an
+      // acceptable length (target is 100-200 characters)
+      // this will return a longer description if one of the first couple of
+      // paragraphs is longer than 200
```
Comment (s-cheng) [inline packages/lesswrong/components/posts/PostsPage/PostsPage.tsx:30]: You don't think it's worth a hard stop at 200 chars?

**Review by s-cheng — APPROVED** [review]
Exciting! ^^

File: packages/lesswrong/components/posts/PostsPage/PostsPage.tsx (hunk 3 above)
Comment (jpaddison3) [reply packages/lesswrong/components/posts/PostsPage/PostsPage.tsx:30]: Aw, let's try it

**Pull request merged**